### PR TITLE
66343

### PR DESCRIPTION
**Pull request by jakaria-istauk**

### CODE REVIEW

File: includes/Template/Eicon-Woocommerce/preset-2.php [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+$sale_badge_preset = isset( $settings['eael_product_sale_badge_preset'] ) ? esc_attr( $settings['eael_product_sale_badge_preset'] ) : '';
+// should print vars
+$should_print_rating = isset( $settings['eael_product_grid_rating'] ) && 'yes' === $settings['eael_product_grid_rating'];
+$should_print_quick_view = isset( $settings['eael_product_grid_quick_view'] ) && 'yes' === $settings['eael_product_grid_quick_view'];
+$should_print_image_clickable = isset( $settings['eael_product_grid_image_clickable'] ) && 'yes' === $settings['eael_product_grid_image_clickable'];
+$should_print_price = isset( $settings['eael_product_grid_price'] ) && 'yes' === $settings['eael_product_grid_price'];
+$should_print_excerpt = isset( $settings['eael_product_grid_excerpt'] ) && ('yes' === $settings['eael_product_grid_excerpt'] && has_excerpt());
+$widget_id = isset($settings['eael_widget_id']) ? $settings['eael_widget_id'] : null;
+
+$sale_badge_text = !empty($settings['eael_product_sale_text']) ? $settings['eael_product_sale_text'] :  __( 'Sale!', 'essential-addons-for-elementor-lite' );
```
Comment (robinwpdeveloper) [inline includes/Template/Eicon-Woocommerce/preset-2.php:48]: `! empty` on all occurances

File: includes/Template/Eicon-Woocommerce/preset-2.php [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+$stock_out_badge_text = !empty($settings['eael_product_stockout_text']) ?$settings['eael_product_stockout_text'] : __( 'Stock <br/> Out', 'essential-addons-for-elementor-lite' );
```
Comment (robinwpdeveloper) [inline includes/Template/Eicon-Woocommerce/preset-2.php:49]: 1 space before and after ternary operator (?).

File: includes/Template/Eicon-Woocommerce/preset-2.php [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+                    <?php
+                    if( $should_print_image_clickable ) {
+                        echo '<a href="' . $product->get_permalink() . '" class="woocommerce-LoopProduct-link woocommerce-loop-product__link">';
+                    }
+                    if ( $is_show_badge ) {
+                        echo( ! $product->is_in_stock() ? '<span class="eael-onsale outofstock ' . esc_attr( $sale_badge_preset . ' ' . $sale_badge_align ) . '">' . $stock_out_badge_text . '</span>' : ( $product->is_on_sale() ? '<span class="eael-onsale ' . esc_attr( $sale_badge_preset . ' ' . $sale_badge_align ) . '">' . $sale_badge_text . '</span>' : '' ) );
+                    }
+                    echo wp_kses_post( $product->get_image( $settings['eael_product_grid_image_size_size'], [ 'loading' => 'eager' ] ) );
+
+                    if( $should_print_image_clickable ) {
```
Comment (robinwpdeveloper) [inline includes/Template/Eicon-Woocommerce/preset-2.php:74]: `if () {}` on all occurances

**Review by robinwpdeveloper — CHANGES_REQUESTED** [review]
Left few feedbacks.
Mostly spacing related.

File: includes/Template/Eicon-Woocommerce/default.php [hunk 4]
```diff
@@ -58,601 +58,49 @@
...
-	                }
-                    do_action( 'eael_woocommerce_after_shop_loop_item' );
-                    ?>
-                </div>
-            </div>
-        </li>
-		<?php
-	}
-}else {
+if($settings['eael_product_grid_rating']!='yes'){
```
[the remark this answers is not in the document]
Comment (jakaria-istauk) [reply includes/Template/Eicon-Woocommerce/default.php:63]: yes

File: includes/Template/Eicon-Woocommerce/default.php [hunk 5]
```diff
@@ -58,601 +58,49 @@
...
+	remove_action('woocommerce_after_shop_loop_item_title','woocommerce_template_loop_rating',5);
+}
 
-	if($settings['eael_product_grid_rating']!='yes'){
-		remove_action('woocommerce_after_shop_loop_item_title','woocommerce_template_loop_rating',5);
+add_action('woocommerce_before_shop_loop_item_title',function() use ( $stock_out_badge_text ){
+	global $product;
+	if ( ! $product->is_in_stock() ) {
+		remove_action( 'woocommerce_before_shop_loop_item_title', 'woocommerce_show_product_loop_sale_flash', 10 );
+		echo '<span class="outofstock-badge">'. $stock_out_badge_text .'</span>';
```
[the remark this answers is not in the document]
Comment (jakaria-istauk) [reply includes/Template/Eicon-Woocommerce/default.php:71]: Although this is a default template and value can changed by user via using hook, if we use escaping it's may causing of design break.

File: includes/Template/Eicon-Woocommerce/default.php [hunk 6]
```diff
@@ -58,601 +58,49 @@
...
-		<?php
-	}
-}else {
+if($settings['eael_product_grid_rating']!='yes'){
+	remove_action('woocommerce_after_shop_loop_item_title','woocommerce_template_loop_rating',5);
+}
 
-	if($settings['eael_product_grid_rating']!='yes'){
-		remove_action('woocommerce_after_shop_loop_item_title','woocommerce_template_loop_rating',5);
+add_action('woocommerce_before_shop_loop_item_title',function() use ( $stock_out_badge_text ){
```
[the remark this answers is not in the document]
Comment (jakaria-istauk) [reply includes/Template/Eicon-Woocommerce/default.php:67]: Although this is a default template and value can changed by user via using hook, if we use escaping it's may causing of design break.

File: includes/Template/Eicon-Woocommerce/default.php [hunk 7]
```diff
@@ -58,601 +58,49 @@
...
-			remove_action( 'woocommerce_before_shop_loop_item_title', 'woocommerce_show_product_loop_sale_flash', 10 );
-			echo '<span class="outofstock-badge">'. $stock_out_badge_text .'</span>';
-		}
-	},9);
-
-	add_filter('woocommerce_sale_flash', function($text, $post, $product) use( $sale_badge_text ) {
-		return '<span class="onsale" data-notification="default">'. $sale_badge_text .'</span>';
-	}, 10, 3);
+add_filter('woocommerce_sale_flash', function($text, $post, $product) use( $sale_badge_text ) {
+	return '<span class="onsale" data-notification="default">'. $sale_badge_text .'</span>';
```
[the remark this answers is not in the document]
Comment (jakaria-istauk) [reply includes/Template/Eicon-Woocommerce/default.php:76]: Although this is a default template and value can changed by user via using hook, if we use escaping it's may causing of design break.

**Review by robinwpdeveloper — APPROVED** [review]

**Pull request merged**